streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -458,7 +474,8 @@ public static WrappedInMemoryWindowStoreIterator emptyIterator() {
 
         private Windowed<Bytes> getWindowedKey() {
             final Bytes key = retainDuplicates ? getKey(super.next.key) : super.next.key;
-            final TimeWindow timeWindow = new TimeWindow(super.currentTime, super.currentTime + windowSize);
+            final long endTime = super.currentTime + windowSize;
+            final TimeWindow timeWindow = new TimeWindow(super.currentTime, endTime < 0 ? Long.MAX_VALUE : endTime);
```

Comment:
Personally, I thought so--but I noticed that's not what we do for the RocksDB version (see WindowKeySchema#timeWindowForSize) and figured there was a reason for it? I'm not too convinced either way but we should definitely be consistent

Reply:
There's a potential KIP for allowing negative timestamps (so you can represent time older than 1970, duh), I think we leave space for such extensions in the future back then.

Reply:
I'm not sure how truncating the window would help with negative timestamps, but I had another thought: was this to allow for a kind of hacky version of UnlimitedWindows before we had support for that (ie user sets window size to Long.MAX)? If so (or otherwise), should we start to throw an exception instead of truncating from here on?